qutip/solver/brmesolve.py

```diff
@@ -351,6 +372,15 @@ def _apply_options(self, keys):
         if need_new_rhs:
             self.rhs = self._prepare_rhs()
 
+        if hasattr(self, "_rhs") and "use_herm_matmul" in keys:
+            self._rhs = self._update_rhs()
+            # Ensure the integrator is reset
+            keys = set(keys)
+            keys.add('method')
+
+        if hasattr(self, "_rhs") and "use_herm_matmul" in keys:
+            self._rhs = self._update_rhs()
```

Comment:
Are these two lines meant to be duplicated here? It seems odd to call `self._update_rhs()` twice.